build.gradle

```diff
@@ -90,6 +90,38 @@ application {
   mainClass = 'net.sourceforge.kolmafia.KoLmafia'
 }
 
+task cleanDist(type: Delete) {
+  onlyIf {
+    file('dist').exists()
+  }
+  inputs.dir('dist')
+  outputs.upToDateWhen { true }
+  doLast {
+    file('dist').eachFile(groovy.io.FileType.FILES) { File file ->
+      if (file.name.startsWith('KoLmafia-') && file.name.endsWith('.jar')) {
+          delete file
+      }
+    }
+  }
+}
+
+task pruneDist(type: Delete) {
+  onlyIf {
+    file('dist').exists()
+  }
+  inputs.dir('dist')
+  outputs.upToDateWhen { true }
+  doLast {
+    file('dist').eachFile(groovy.io.FileType.FILES) { File file ->
+      if (file.name.startsWith('KoLmafia-') && file.name.endsWith('.jar')) {
```

Comment:
Two questions: is it case sensitive?  I think that's ok if it is.

And also unimportantly, can we use .startsWith('KoLmafia') so that users who do ANT builds can get cleaned up, too?

Reply:
It should be case-sensitive.

Don't Ant builds specify a dash, either `daily` or `jar`?

Reply:
Nope.  It comes straight from sVN:
```

Michaels-MBP:kolmafia mcroft$ svnversion
20947M
```

I didn't fight it at the time because I didn't care that much and it gives me a visual clue as to which version we used.

Reply:
That's the version, not the full jar. Right?

```
version-name=${ant.project.name}-${version}
jarfile=${dist}/${version-name}.jar
```

My jars that I built just now with Ant look like KoLmafia-25704.jar and KoLmafia-21.9.jar.

Reply:
Oh, I see, this probably should be endsWith('M.jar') without the dash. That's what you're getting at, right?

Reply:
Yeah. Ant is firmly *not supported* at this point, given that it can't even bundle revision numbers anymore.

Reply:
Yeah, it's got a few features we don't have in gradle, but it's not something we want to support.

daily is probably the biggest missing item, which depends on ANT calling out to svn update.

We can switch that to grgit which relies on JGit, which I want to use anyway for git scripts, but it's a bigger task than I thought at first.